src/librbd/cache/pwl/AbstractWriteLog.h

```diff
@@ -399,7 +400,7 @@ class AbstractWriteLog {
   virtual uint64_t get_max_extent() {
     return 0;
   }
-
+  void update_image_cache_state(void);
```

Comment:
> Move this line before `void update_image_cache_state(Context *on_finish);` so that the overload and `handle_update_image_cache_state` declarations are all close to each other.

this line need to protected, but the `void update_image_cache_state(Context *on_finish)` is originally private.  I moved these three here as protected.

Reply:
Sorry, I missed that.  Disregard my earlier comment and move those back -- let's not make methods that can be private non-private.